### PR TITLE
Copy attributes back post de-serializing array

### LINKED ISSUE #450: Array flattening in deserializer loses previously de-serialized attributes
Referencing the following:
https://github.com/Azure/ms-rest-js/blob/ef37d563d7f604cab8a151c8cc50f064bddfc1b4/lib/serializer.ts#L738

This code blows away any previously deseriaized attributes on the instance. This causes a loss of information such as the odata.* attributes in the following model. 

```
    var PolicyEventsQueryResults = {
        serializedName: "PolicyEventsQueryResults",
        type: {
            name: "Composite",
            className: "PolicyEventsQueryResults",
            modelProperties: {
                odatacontext: {
                    serializedName: "@odata\\.context",
                    type: {
                        name: "String"
                    }
                },
                odatacount: {
                    serializedName: "@odata\\.count",
                    constraints: {
                        InclusiveMinimum: 0
                    },
                    type: {
                        name: "Number"
                    }
                },
                odatanextLink: {
                    serializedName: "@odata\\.nextLink",
                    type: {
                        name: "String"
                    }
                },
                value: {
                    serializedName: "",
                    type: {
                        name: "Sequence",
                        element: {
                            type: {
                                name: "Composite",
                                className: "PolicyEvent",
                                additionalProperties: {
                                    type: {
                                        name: "Object"
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
```

A short-term work-around is to move the value property to the start of the model, but a proper fix would be to either sort the order of properties being parsed to do the array generation first, or to save and re-apply any existing attributes.

### PR DESCRIPTION
**Pull request by mgranell**
Fixes #450

### CODE REVIEW

File: lib/serializer.ts [hunk 1]
```diff
@@ -735,7 +735,15 @@ function deserializeCompositeType(
       // paging
       if (Array.isArray(responseBody[key]) && modelProps[key].serializedName === "") {
         propertyInstance = responseBody[key];
-        instance = serializer.deserialize(propertyMapper, propertyInstance, propertyObjectName);
+        const arrayInstance = serializer.deserialize(propertyMapper, propertyInstance, propertyObjectName);
+        // Copy over any properties that have already been added into the instance, where they do
+        // not exist on the newly de-serialized array
+        Object.entries(instance).forEach(([key, value]) => {
```
Comment (xirzec) [inline lib/serializer.ts:741]: nit
```suggestion
        for (const [key, value] of Object.entries(instance)) {
```

**Comment by mgranell** [comment]
> nit
```suggestion
        for (const [key, value] of Object.entries(instance)) {
```

File: lib/serializer.ts (hunk 1 above)
Comment (mgranell) [reply lib/serializer.ts]: Thanks, updated

**Comment by ramya-rao-a** [comment]
@jeremymeng, @xirzec Do we have a similar issue in `@azure/core-http` or the newer Azure Core packages?

**Comment by jeremymeng** [comment]
> @jeremymeng, @xirzec Do we have a similar issue in `@azure/core-http` or the newer Azure Core packages?

I am looking into it.

**Review by jeremymeng — APPROVED** [review]

**Comment by jeremymeng** [comment]
@deyaaeldeen @xirzec are you good with the fix?

**Review by xirzec — APPROVED** [review]

**Pull request merged**